### PR TITLE
feat: No expression error when node hasn’t executed

### PR DESCRIPTION
**Pull request by elsmr**
## Summary
image

When a Node has no input data:
- Expressions that read input data should be grey (instead of red) + should have no error message
- When focused, the output should show a helpful message: "Test previous nodes for preview"

## Related tickets and issues
https://linear.app/n8n/issue/NODE-1075/no-expression-error-when-node-hasnt-executed

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

**Review by netroy — DISMISSED** [review]
Changes in `n8n-workflow` LGTM 🎉

File: cypress/e2e/14-mapping.cy.ts [hunk 1]
```diff
@@ -181,11 +181,6 @@ describe('Data mapping', () => {
 		ndv.getters
 			.inlineExpressionEditorInput()
 			.should('have.text', `{{ $('${SCHEDULE_TRIGGER_NODE_NAME}').item.json.input[0].count }}`);
-		ndv.getters
-			.parameterExpressionPreview('value')
-			.invoke('text')
-			.invoke('replace', /\u00a0/g, ' ')
-			.should('equal', '[ERROR: no data, execute "Schedule Trigger" node first]');
```
Comment (elsmr) [inline cypress/e2e/14-mapping.cy.ts:188]: This is no longer shown as an error

**Review by MiloradFilipovic — APPROVED** [review]
Front-end looks good to me

**Comment by janober** [comment]
Got released with [`n8n@1.31.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.31.0)

**Pull request merged**